server/game/cards/06-CotE/UnveiledDestiny.js

```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class UnveiledDestiny extends DrawCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            condition: context => context.source.parent.isAttacking() && context.player.role.getElement() !== [],
```

Comment:
I don't think this will work due to the way js deals with Arrays.  `getElement()` will return a reference (or pointer) to an Array, and `[]` will be a reference to a different Array, so this `!==` will always evaluate as `true` even when both arrays are empty.  However, due to the way you've done the `effect` below, I think you can just remove this altogether. You should check that the player does actually have a role card here though, so I would put `condition: context => context.player.role && context.source.parent.isAttacking()`